dashbuilder/dashbuilder-backend/dashbuilder-services/src/test/java/org/dashbuilder/DashbuilderMigrationServiceTest.java

```diff
@@ -0,0 +1,230 @@
+            add("/readme.md");
+        }});
+    }
+
+    @Test
+    public void testMigratePerspectives() {
+        checkInitialCondition();
+
+        createFiles(sourceFS);
+
+        migrationService.migratePerspectives(sourceFS, targetFS);
+
+        List<String> sourceFiles = getFiles(sourceFS);
+        List<String> targetFiles = getFiles(targetFS);
+
+        assertEquals(sourceFiles.size(), 6);
+        assertEquals(targetFiles.size(), 3);
+        assertEquals(sourceFiles, new ArrayList<String>() {{
+            add("/bbb.txt");
+            add("/dataset1.csv");
+            add("/definitions/dataset2.dset");
+            add("/navtree.json");
+            add("/perspective_layouts/aaa.txt");
+            add("/readme.md");
+        }});
+        assertEquals(targetFiles, new ArrayList<String>() {{
+            add("/page-abc/perspective_layout.json");
+            add("/perspective_layout2.txt");
+            add("/readme.md");
+        }});
+    }
+
+    @Test
+    public void testMigrateNavigation() {
+        checkInitialCondition();
+
+        createFiles(sourceFS);
+
+        migrationService.migrateNavigation(sourceFS, targetFS);
+
+        List<String> sourceFiles = getFiles(sourceFS);
+        List<String> targetFiles = getFiles(targetFS);
+
+        assertEquals(sourceFiles.size(), 7);
+        assertEquals(targetFiles.size(), 2);
+        assertEquals(sourceFiles, new ArrayList<String>() {{
+            add("/bbb.txt");
+            add("/dataset1.csv");
+            add("/definitions/dataset2.dset");
+            add("/page-abc/perspective_layout.json");
+            add("/perspective_layout2.txt");
+            add("/perspective_layouts/aaa.txt");
+            add("/readme.md");
+        }});
+        assertEquals(targetFiles, new ArrayList<String>() {{
+            add("/navtree.json");
+            add("/readme.md");
+        }});
+    }
+
+    private void checkInitialCondition() {
+        List<String> sourceFiles = getFiles(sourceFS);
+        List<String> targetFiles = getFiles(targetFS);
+
+        assertEquals(sourceFiles.size(), 1);
+        assertEquals(targetFiles.size(), 1);
+        assertEquals(sourceFiles, new ArrayList<String>() {{
+            add("/readme.md");
+        }});
+        assertEquals(targetFiles, new ArrayList<String>() {{
+            add("/readme.md");
+        }});
+    }
+
+    private List<String> getFiles(FileSystem fs) {
+        List<String> files = new ArrayList<>();
+        Path root = fs.getRootDirectories().iterator().next();
+
+        Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(final Path path, final BasicFileAttributes attrs) throws IOException {
+                files.add(path.toString());
+                 return FileVisitResult.CONTINUE;
+            }
+        });
+
+        return files;
+    }
+
+    private void createFiles(FileSystem fs) {
```

Comment:
Please make sure that the files are deleted after the test finishes.

Reply:
Done!